core/src/main/java/org/apache/calcite/sql/type/OperandTypes.java

```diff
@@ -216,6 +216,11 @@ public static SqlOperandTypeChecker variadic(
   public static final SqlSingleOperandTypeChecker BOOLEAN_BOOLEAN =
       family(SqlTypeFamily.BOOLEAN, SqlTypeFamily.BOOLEAN);
 
+  public static final SqlSingleOperandTypeChecker NUMERIC_BOOLEAN_BOOLEAN =
+      family(ImmutableList.of(SqlTypeFamily.NUMERIC, SqlTypeFamily.BOOLEAN, SqlTypeFamily.BOOLEAN),
+          //2nd and 3rd operands are optional
+          i -> i == 1 || i == 2);
```

Comment:
Please check is this condition is required here. 
Supported operand types  for the functions are already mentioned as either Numeric or Numeric_boolean_boolean, implying that 2nd and 3rd operands are optional.
Might work without this condition too.

Reply:
yes checked with that and its working so removing that changes